cocos/2d/CCSprite.cpp

```diff
@@ -628,37 +673,41 @@ void Sprite::setTextureCoords(const Rect& rectInPoints, V3F_C4B_T2F_Quad* outQua
         return;
     }
 
-    auto rectInPixels = CC_RECT_POINTS_TO_PIXELS(rectInPoints);
+    const auto rectInPixels = CC_RECT_POINTS_TO_PIXELS(rectInPoints);
 
-    float atlasWidth = (float)tex->getPixelsWide();
-    float atlasHeight = (float)tex->getPixelsHigh();
+    const float atlasWidth = (float)tex->getPixelsWide();
+    const float atlasHeight = (float)tex->getPixelsHigh();
 
-    float left, right, top, bottom;
+    float rw = rectInPixels.size.width;
+    float rh = rectInPixels.size.height;
 
     if (_rectRotated)
-    {
+        std::swap(rw, rh);
+
 #if CC_FIX_ARTIFACTS_BY_STRECHING_TEXEL
-        left    = (2*rectInPixels.origin.x+1) / (2*atlasWidth);
-        right   = left+(rectInPixels.size.height*2-2) / (2*atlasWidth);
-        top     = (2*rectInPixels.origin.y+1) / (2*atlasHeight);
-        bottom  = top+(rectInPixels.size.width*2-2) / (2*atlasHeight);
+    float left    = (2*rectInPixels.origin.x+1) / (2*atlasWidth);
+    float right   = left+(rw*2-2) / (2*atlasWidth);
+    float top     = (2*rectInPixels.origin.y+1) / (2*atlasHeight);
+    float bottom  = top+(rh*2-2) / (2*atlasHeight);
 #else
-        left    = rectInPixels.origin.x / atlasWidth;
-        right   = (rectInPixels.origin.x + rectInPixels.size.height) / atlasWidth;
-        top     = rectInPixels.origin.y / atlasHeight;
-        bottom  = (rectInPixels.origin.y + rectInPixels.size.width) / atlasHeight;
+    float left    = rectInPixels.origin.x / atlasWidth;
+    float right   = (rectInPixels.origin.x + rw) / atlasWidth;
+    float top     = rectInPixels.origin.y / atlasHeight;
+    float bottom  = (rectInPixels.origin.y + rh) / atlasHeight;
 #endif // CC_FIX_ARTIFACTS_BY_STRECHING_TEXEL
 
-        if (_flippedX)
-        {
-            std::swap(top, bottom);
-        }
+    if ((_flippedX && !_rectRotated) || (_flippedY && _rectRotated))
+    {
+        std::swap(left, right);
```

Comment:
No new logic was added in this code. I just "compressed" a little bit the code in order to avoid having duplicated code (less error prone).
As you can see from here ( https://github.com/cocos2d/cocos2d-x/blob/cocos2d-x-3.13.1/cocos/2d/CCSprite.cpp#L483 ) there was a big `if (_rectRotated)` and the code was almost a duplicate in both guards